src/HealthChecks.AzureServiceBus/AzureServiceBusQueueHealthCheck.cs

```diff
@@ -1,40 +1,57 @@
-using Azure.Core;
+using HealthChecks.AzureServiceBus.Configuration;
 using Microsoft.Extensions.Diagnostics.HealthChecks;
 
 namespace HealthChecks.AzureServiceBus
 {
     public class AzureServiceBusQueueHealthCheck : AzureServiceBusHealthCheck, IHealthCheck
     {
-        private readonly string _queueName;
+        private readonly AzureServiceBusQueueHealthCheckOptions _options;
         private string? _connectionKey;
 
-        public AzureServiceBusQueueHealthCheck(string connectionString, string queueName) : base(connectionString)
-        {
-            _queueName = Guard.ThrowIfNull(queueName, true);
-        }
+        protected override string ConnectionKey => _connectionKey ??= $"{Prefix}_{_options.QueueName}";
 
-        public AzureServiceBusQueueHealthCheck(string endPoint, string queueName, TokenCredential tokenCredential) :
-            base(endPoint, tokenCredential)
+        public AzureServiceBusQueueHealthCheck(AzureServiceBusQueueHealthCheckOptions options)
+            : base(options)
         {
-            _queueName = Guard.ThrowIfNull(queueName, true);
+            Guard.ThrowIfNull(options.QueueName, true);
+
+            _options = options;

```

Comment:
Base class contains the same options instance in `_options` field. I suggest to make it `protected` to be able ti use in derived classes (with cast of course - either inline or via private property).

Reply:
I'm fine to merge after you fix duplicated options field.